app/components/pages/BrowseOverlays.vue.ts

```diff
@@ -28,34 +26,31 @@ export default class BrowseOverlays extends Vue {
   @Inject() scenesService: ScenesService;
   @Inject() private notificationsService: NotificationsService;
   @Inject() private jsonrpcService: JsonrpcService;
-  @Inject() private i18nService: I18nService;
 
   @Prop() params: {
     type?: 'overlay' | 'widget-theme';
     id?: string;
   };
 
-  $refs: {
-    overlaysWebview: Electron.WebviewTag;
-  };
-
-  mounted() {
-    this.$refs.overlaysWebview.addEventListener('did-finish-load', () => {
-      this.guestApiService.exposeApi(this.$refs.overlaysWebview.getWebContents().id, {
+  onBrowserViewReady(view: Electron.BrowserView) {
+    view.webContents.on('did-finish-load', () => {
+      this.guestApiService.exposeApi(view.webContents.id, {
         installOverlay: this.installOverlay,
         installWidgets: this.installWidgets,
       });
     });
 
-    this.$refs.overlaysWebview.addEventListener('new-window', e => {
-      const protocol = urlLib.parse(e.url).protocol;
+    electron.ipcRenderer.send('webContents-preventPopup', view.webContents.id);
+
+    view.webContents.on('new-window', (e, url) => {
+      const protocol = urlLib.parse(url).protocol;
 
       if (protocol === 'http:' || protocol === 'https:') {
-        electron.remote.shell.openExternal(e.url);
+        electron.remote.shell.openExternal(url);
       }
     });
 
-    I18nService.setWebviewLocale(this.$refs.overlaysWebview);
+    I18nService.setBrowserViewLocale(view);
```

Comment:
is there a reason you didn't just use the `setLocale` option here?

Reply:
Good catch - I added the `setLocale` prop the second time I needed it, and forgot to refactor the initial place.